### DISCUSSION TITLE
Enabling Debug Logging

### DISCUSSION THREAD

**Discussion by ollien** [vscode-neovim/vscode-neovim#1952] [Q&A] [answered]
I've struggled to get debug logs to work when working on the extension. I tried doing "Developer: Set Log Level" > "vscode-neovim" > "Trace", but I don't see anything in the output console (with vscode-neovim selected), or the developer console. Is there something else that I'm meant to do?

**Comment by justinmk** [comment, marked as answer]
We *should* update the codebase to do what you are expecting. But currently the extension log-level is driven by the `vscode-neovim.logLevel` setting.

This is because vscode `LogOutputChannel` was only recently implemented: https://github.com/vscode-neovim/vscode-neovim/commit/f77256be6afcaa596dc8c3cb9d4e6bc0cb12cfab

Note: there is also an outdated comment here: https://github.com/vscode-neovim/vscode-neovim/blob/4ccd611e09f6b5538f281920b3c6a27738a13511/src/logger.ts#L112-L118

# Proposal

We should remove the `vscode-neovim.logLevel` setting and let `Developer: Set Log Level` (and the local `LogOutputChannel.logLevel` field) decide the vscode-neovim log level.

**Comment by theol0403** [comment]
I also believe when the log channel was implemented, there was no way to set the log level in vscode.
